docs/design/manipulation/analysis.md

```diff
@@ -0,0 +1,195 @@
+      services to an external orchestration tool) or actively being (using
+      native O3DE scripts to drive the manipulation course of actions).
+
+    * Manipulation planning: integral manipulation planning could be delegated
+      to third party libraries (such as MoveIt) or parts of the planning to
+      external planners (such as OMPL). In both cases O3DE will probably need
+      to provide information about the state of joints, the state of the scene,
+      and other data/metadata about the state of the whole simulaton.
+
+ * Motion execution: having the ability to use both, internal native O3DE
+   motion controllers, or external controllers implemented in frameworks
+   or libraries (such as ros2_control) is an important part of the
+   manipulation pipeline. Note that the use of controllers is optional
+   for quick prototyping so the manipulation planning could interact
+   directly with the simulator without using a controller.
+
+Diagram of input/outpus and data flows using ROS 2:
+
+![Manipulation diagram](manipulation.svg)
+
+To implement these set of actions, the O3DE simulator should have at least:
+
+ * The capacity of publish information about the state of the joints (**RO2JointStatePub** component):
+   Include the needed information from O3DE simulation related to the joints of
+   the simulated robot in `sensor_msgs/JointState.msg` and publish it in the `/joint_states`
+   topic. This is required to interact with ROS 2 packages (such as MoveIt)
+   See [gazebo_ros_pkgs plugin for reference](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/galactic/gazebo_plugins/include/gazebo_plugins/gazebo_ros_joint_state_publisher.hpp#L44-L73)
+
+ * Receive orders to move the different joints (**ROS2PoseTrajectory** component):
+   Set the trajectory of points to be followed by joints in the O3DE simulation using as input
+   `trajectory_msgs/msg/JointTrajectory.msg`
+   See [gazebo_ros_pkgs plugin for reference](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/galactic/gazebo_plugins/include/gazebo_plugins/gazebo_ros_joint_pose_trajectory.hpp#L26-L48)
+
+ * The ability to request picks or motion movements to external manipulation planners (**ROS2ManipulationRequester**):
+   Request a motion plan to external planners given different types of inputs,
+   a target pose for end effector or a joint state target. MoveIt `move_group_interface` interaction
+   is an example of this.
+   See [MoveIt 2 joint state target goal](https://github.com/ros-planning/moveit2/blob/main/moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h#L305-L320)
+   or [MoveIt 2](https://github.com/ros-planning/moveit2/blob/main/moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h#L640-L554)
+
+ * Provide data from the simulated sensors: not in the scope of this document
```

Comment:
> I am strongly in favor of moving this section to the architecture.md document.

I have nothing against it and can happily accept the patch. The only reason why I put it here at the begging is because I tried to follow the [vehicle_dynamics](https://github.com/RobotecAI/o3de-ros2-gem/tree/development/docs/design/vehicle_dynamics) docs that were the first one in the design dir.

@adamdbrw do you have any preference?